packages/core/src/fhir-to-cda/cda-templates/components/component-of.ts

```diff
@@ -0,0 +1,119 @@
+import { Bundle, Encounter, Location, Practitioner } from "@medplum/fhirtypes";
+import { toArray } from "@metriport/shared";
+import {
+  findCompositionResource,
+  findResourceInBundle,
+  isEncounter,
+} from "../../../external/fhir/shared";
+import {
+  buildAddress,
+  buildCodeCe,
+  buildCodeCvFromCodeCe,
+  buildInstanceIdentifier,
+  buildTelecom,
+  formatDateToCdaTimestamp,
+  withoutNullFlavorObject,
+} from "../commons";
+import { placeholderOrgOid } from "../constants";
+import { AugmentedEncounter } from "./augmented-resources";
+import { createAugmentedEncounters } from "./encounters";
+import { EncompassingEncounter, ResponsibleParty } from "../../cda-types/shared-types";
+
+function findEncounterInBundle(
+  fhirBundle: Bundle,
+  encounterId: string | undefined
+): Encounter | undefined {
+  if (!encounterId) return undefined;
+  const resource = findResourceInBundle(fhirBundle, encounterId);
+  return isEncounter(resource) ? resource : undefined;
+}
+
+export function buildComponentOf(fhirBundle: Bundle): EncompassingEncounter | undefined {
```

Comment:
I Was a little confused with the naming here since we are actually building an EncompassingEncounter right? 

Also can we get a unit test + CDA example for what this section looks like the same way we have done for all the others?

Reply:
Yea, I guess the naming / type need adjustment, so they match each other. But, technically, we're building the `componentOf` attribute of a ClinicalDocument..

Reply:
Fixed